src/lintsampler/lintsampler.py

```diff
@@ -0,0 +1,547 @@
+            inferred dimensionality of the problem.
+                
+        """
+        self._setgrid(cells)
+
+
+    def _setgrid(self,cells):
+        """Construct the cells for sampling.
+
+        Parameters
+        ----------
+        self : LintSampler
+            The LintSampler instance.
+        cells : single array, tuple of arrays or list of tuples of arrays
+            If a single array, the boundary values for a 1D grid. If a tuple of arrays, the
+            boundary values for a grid with dimensionality of the length of the tuple. If
+            a list of tuples of arrays, the boundary values for an arbitrary number of
+            grids with dimensionality the length of the tuples.
+
+        Returns
+        -------
+        None
+        
+        Attributes
+        -------
+        eval_type : string
+            The evaluation type, either 'gridsample' or 'freesample'
+        dim : int
+            The inferred dimensionality of the problem
+        edgearrays : list of numpy arrays
+            The arrays defining the edge of the grids
+        edgedims   : list of integers, optional, used if eval_type == 'gridsample'
+            The shape of the gridsample grid
+        ngrids : integer, optional, used if eval_type == 'freesample'
+            The number of distinct grids to be constructed
+        gridshape : list of tuple of integers, optional, used if eval_type == 'freesample'
+            The shape of each distinct grid
+        ngridentries : list of integers, optional, used if eval_type == 'freesample'
+            The number of entries in the 
+        nedgegridentries : list of integers, optional, used if eval_type == 'freesample'
+            The number of entries in the offset grids
+        x0 : list of numpy arrays, optional, used if eval_type == 'freesample'
+            The array of the first vertex of the grids
+        x1 : list of numpy arrays, optional, used if eval_type == 'freesample'
+            The array of the last vertex of the grids
+
+        
+        """
+
+        # check the validity of the input
+        if len(cells)==0:
+            raise ValueError("LintSampler._setgrid: you must specify an evaluation domain with a single array, a tuple of arrays or a list of tuples of arrays. See documentation for details.")
+            # or, we could here drop into an adaptive grid selection
+            # or, we could sample on a unit hypercube with dimensions of the pdf
+
+        # 0. are we in the 1d case? the input is just a single tuple or 1d array.
+        # i.e. cells = np.linspace(-12,12,100)
+        if hasattr(cells, "__len__") and not hasattr(cells[0], "__len__"):
+            self.eval_type='gridsample'
+
+            # override the inferred dimensionality
+            self.dim = 1
+
+            # set the arrays
+            # we are being passed the array directly
+            self.edgearrays = np.array(cells)
```

Comment:
The code in `sample` (lines 227 to 241) can be simplified a little if `edgearrays` is always a list, even in the 1D case.
```suggestion
            self.edgearrays = [np.array(cells)]
```